ebean-core/src/main/java/io/ebeaninternal/api/BindHash.java

```diff
@@ -0,0 +1,37 @@
+package io.ebeaninternal.api;
+
+/**
+ * BindHash implementation.
+ *
+ * @author Roland Praml, FOCONIS AG
+ *
+ */
+public interface BindHash {
+
+  /**
+   * Update with boolean value.
+   */
+  BindHash update(boolean boolValue);
+
+  /**
+   * Update with int value.
+   */
+  BindHash update(int intValue);
+
+  /**
+   * Update with long value.
+   */
+  BindHash update(long longValue);
+
+  /**
+   * Update with object value.
+   */
+  BindHash update(Object value);
+
+  /**
+   * finishes the hash. May be used to compute internal state. After finish, no
+   * update method must be called
+   */
+  void finish();
+
```

Comment:
As HashCodeBindHash is just a holder for all values, we could think to change the `public interface BindHash`
to 
```java
public class BindHash {
   private final List<Object> values = new ArrayList<>();
   public void update(Object value) { values.add(value); }
   public int hashCode() { return values.hashCode; }
   public boolean equals(Object o) { ... }
}
```
Is `BindHash` still the right word. 

BTW: I like the idea, to store the object values, and remove the MdBindHash as it is impossible now to get a collision. 

But do we have to pay attention, so that we do not run in an OOM problem when very long strings/big byte arrays ... are used?
(We had the problem with the serverCache, thats why we made 
https://github.com/FOCONIS/ebean/commit/44193f36e5d1c9c67ac50ada9193b06e8cda7425 - pr will come)

Reply:
> problem with the serverCache, thats why we made

Yeah, nice !!